lib/registry-secrets.ts

```diff
@@ -74,14 +74,15 @@ export class RegistrySecretValidator {
 	 * @returns The input object cast to the RegistrySecrets type if validation succeeds
 	 * @throws Throws an error if validation fails
 	 */
-	public validateRegistrySecrets(parsedJson: object): RegistrySecrets {
+	public validateRegistrySecrets(
+		parsedJson: unknown,
+	): asserts parsedJson is RegistrySecrets {
 		const valid = this.validateFunction(parsedJson);
 		if (!valid) {
 			throw new RegistrySecretValidationError(
 				this.validator.errorsText(this.validateFunction.errors),
 			);
 		}
-		return parsedJson as RegistrySecrets;
```

Comment:
Removing this return statement is a breaking change that requires `Change-type: major`, while this PR and commits are declared `Change-type: minor`. For example, the CLI relies on this return value:
https://github.com/balena-io/balena-cli/blob/v12.14.18/lib/utils/compose_ts.ts#L477

It's OK to have a breaking change and `Change-type: major`, but also it would be OK to simply leave the return statement there. :-)

Reply:
not sure why this one was removed. will re-add it rather than break api

Reply:
this slipped in here btw: https://github.com/balena-io-modules/resin-multibuild/pull/81/files#diff-af4b6a79d80562b9ccd9f13d18594741L84